packages/data/src/index.js

```diff
@@ -297,9 +303,13 @@ export const withSelect = ( mapStateToProps ) => createHigherOrderComponent( ( W
 			return this.shouldComponentUpdate;
 		}
 
-		componentWillReceiveProps( nextProps ) {
```

Comment:
This changes could be impactful because of the order between the lifecycle calls, but in my testing, it's working as expected.